app/__main__.py

```diff
@@ -3,11 +3,57 @@
 import discord
 import dotenv
 from discord import app_commands
+from discord.ui import Button, View
 
 dotenv.load_dotenv()
 TOKEN = os.environ["TOKEN"]
 
 
+class UpgradeView(View):
+    """A discord.View subclass to handle user interactions with the update screen."""
+
+    def __init__(self) -> None:
+        super().__init__(timeout=None)
+
+    @discord.ui.button(label="Upgrade CPS", style=discord.ButtonStyle.blurple, custom_id="upgradepersistent:cps")
+    async def cps_upgrade(self, interaction: discord.Interaction, button: Button) -> None:
+        """Upgrade a user's CPS."""
+        button.label = "CPS Selected"
+        if interaction.message is not None:
+            await interaction.message.edit(view=self)
+        await interaction.response.send_message(
+            "Dummy message confirming CPS method works",
+            ephemeral=True,
+        )
+
+    @discord.ui.button(
+        label="Upgrade Priority",
+        style=discord.ButtonStyle.blurple,
+        custom_id="upgradepersistent:priority",
+    )
+    async def priority_upgrade(self, interaction: discord.Interaction, button: Button) -> None:
+        """Upgrade a user's priority."""
+        button.label = "Priority Selected"
+        if interaction.message is not None:
+            await interaction.message.edit(view=self)
+        await interaction.response.send_message(
+            "Dummy message confirming priority upgrade method works",
+            ephemeral=True,
+        )
+
+    @discord.ui.button(label="Cancel", style=discord.ButtonStyle.red, custom_id="upgradepersistent:cancel")
+    async def cancel(self, interaction: discord.Interaction, button: Button) -> None:
+        """Cancel the upgrade process."""
+        button.label = "Cancelled"
+        self.stop()
```

Comment:
This doesn't make sense for persistent views

Reply:
We *could* make all buttons disabled but I think cancelling persistent views is unnecessary.